rw-system.sh

```diff
@@ -529,12 +529,32 @@ if getprop ro.boot.boot_devices |grep -v , |grep -qE .;then
 fi
 
 if [ -c /dev/dsm ];then
+    # /dev/dsm is a magic device on Kirin chipsets that teecd needs to access.
+    # Make sure that permissions are right.
     chown system:system /dev/dsm
     chmod 0660 /dev/dsm
+
+    # The presence of /dev/dsm indicates that we have a teecd,
+    # which needs /sec_storage and /data/sec_storage_data
+
     mkdir -p /data/sec_storage_data
     chown system:system /data/sec_storage_data
-    chcon u:object_r:teecd_data_file_system:s0 /data/sec_storage_data
-    mount /data/sec_storage_data /sec_storage
+    chcon -R u:object_r:teecd_data_file:s0 /data/sec_storage_data
+
+    mount | grep " on /sec_storage " > /dev/null 2>&1
+    if [ "$?" -eq "0" ]; then
```

Comment:
`if mount | grep -q " on /sec_storage ";then`

Reply:
Thanks for the hint, fixed.